demos/sjtwo/eeprom/source/main.cpp

```diff
@@ -0,0 +1,50 @@
+#include <project_config.hpp>
+
+#include <cstdint>
+#include <iterator>
+
+#include "L1_Peripheral/lpc40xx/eeprom.hpp"
+#include "L2_HAL/displays/led/onboard_led.hpp"
+#include "utility/log.hpp"
+#include "utility/time.hpp"
+
+int main(void)
+{
+  LOG_INFO("Starting EEPROM Example");
+  constexpr size_t kPayloadSize = 128;
+  constexpr uint32_t kAddress   = 0b0110'0011'1000;
+
+  uint8_t * results = new uint8_t[kPayloadSize];
+
+  LOG_INFO("Initializing EEPROM");
+  sjsu::lpc40xx::Eeprom eeprom;
+  eeprom.Initialize();
+
+  // Generating Payload
+  uint8_t * list = new uint8_t[kPayloadSize];
+
+  for (size_t i = 0; i < kPayloadSize; i++)
```

Comment:
Its more direct and readable if you check the size of the array like so. But it requires that the size be knowable at compile time, which requires that you move away from heap allocation.
```suggestion
  for (size_t i = 0; i < std::size(list); i++)
```